### DISCUSSION TITLE
Importing STL file using EB Build incflo

### DISCUSSION THREAD

**Discussion by brahmandas** [AMReX-Codes/amrex#3642] [Q&A]
I am trying to use AMReX's STL functionality with incflo. I am calling `EB::build` like this 

`EB2::Build(geom.back(), 0, 20)`

This works OK for me with amr.max_level =0 but gives me segfault when I use max_level > 0. It fails when it tries to build the level from scratch. I am testing this with the airfoil.stl provided with amrex EB tutorial. Could you please suggest me what I might be doing wrong? I appreciate any help or insight given in this issue.

**Comment by WeiqunZhang** [comment]
Try `EB2::Build(geom.back(), max_level, max_coarsening_level)`, where `max_level` is amr.max_level, and `max_coarsening_level` is 20 if you use amrex's linear solvers, max_level otherwise. You probably might get an assertion error if the resolution is not fine enough.

For debugging purpose, try `USE_ASSERTION=TRUE` for GNU make and `-DAMReX_ASSERTIONS=YES` for CMake.

**Reply by brahmandas** [reply]
Hello @WeiqunZhang, Thank you for your reply. I'd like to confirm regarding your answer, does that mean that in the `EB2::Build(geom.back(), max_level, max_coarsening_level)` we need to specify the `max_level` as the amr max level we want to achieve in the simulation case?
As you suggested, I tried changing 0 to max_level in `EB2::Build(geom.back(), 0, 20)`, however, it gives segfault for higher amr levels. Do you think it could be the STL itself? In this case, I was trying to simulate the airfoil STL provided with the tutorials.
I appreciate your help and looking forward to your reply.

**Reply by WeiqunZhang** [reply]
It might help, if you could provide a small reproducer. If not, could you try to build and run in DEBUG mode, and provide more information such as the backtrace files?

**Reply by brahmandas** [reply]
Thanks @WeiqunZhang for your help. This is how I am creating the geometry from the stl
`void incflo::make_eb_stl()`
`{
    // Build index space`
    `int max_level_here = 2;`
    `int max_coarsening_level = 20;`
   `EB2::Build(geom.back(), max_level_here, max_level_here + max_coarsening_level);
}`
I am currently hard coding the max_level value. I am not sure how to get this value from the input file inside this function.
I am also attaching the stl file I am using and the input file.  For this case if I use the grid size (64 8 8 ) I can build the EBGeometry and run the simulation with max_level 2, but if I refine the base mesh to ( 128 16 16 ) I get segfault. I compiled the code with Debug and ASSERTION true. I have attached the log and the Backtrace. The same happens if I keep the base grid size to (64 8 8 ) and increase the max_level to 3 (after changing max_level to 3 in eb_stl and recompiling the code)
[attachments.zip](https://github.com/AMReX-Codes/amrex/files/13893085/attachments.zip)
